apps/prairielearn/src/lib/instructorFiles.js

```diff
@@ -163,5 +166,23 @@ export function getPaths(req, res, callback) {
       }
     });
 
-  callback(null, paths);
+  return paths;
+}
+
+/**
+ * Wrapper around {@link getPaths} to support callback-based usage.
+ *
+ * @param {import('express').Request} req
+ * @param {import('express').Response} res
+ * @param {(err: Error | null | undefined, paths?: any) => void} callback
+ */
+export function getPathsCallback(req, res, callback) {
+  let paths;
+  try {
+    paths = getPaths(req, res);
+  } catch (err) {
+    callback(err);
+    return;
+  }
+  callback(paths);
```

Comment:
```suggestion
  callback(null, paths);
```

Reply:
😓 at least the tests failed correctly...